src/include/txn.i

```diff
@@ -857,7 +857,7 @@ __wt_txn_read_upd_list(
          * process the stop time window to decide if there was a tombstone on the update chain.
          */
         if (type == WT_UPDATE_TOMBSTONE && F_ISSET(&cbt->iface, WT_CURSTD_IGNORE_TOMBSTONE) &&
-          !__wt_txn_upd_visible_all(session, upd)) {
+          !WT_TIME_WINDOW_HAS_STOP(&cbt->upd_value->tw)) {
```

Comment:
Can you add a comment here to explain why we should check for whether it has already a stop timestamp or not? This is because of multiple tombstones, so we want to return only the most recent one. Adding a comment would be good.